TauAnalysis/MCEmbeddingTools/plugins/TrackMergeremb.cc

```diff
@@ -0,0 +1,391 @@
+        outTracks_exgsf->push_back( reco::GsfTrackExtra( *it->gsfExtra() ) );
+        
+        outTracks->back().setExtra( reco::TrackExtraRef( rTrackExtras, outTracks_ex->size() - 1) );
+        outTracks->back().setGsfExtra( reco::GsfTrackExtraRef( rTrackExtras_gsf, outTracks_exgsf->size() - 1) );
+        }
+	          
+    }// end merge 
+    
+    
+    iEvent.put(std::move(outTracks),instance);
+    iEvent.put(std::move(outTracks_ex),instance);
+    iEvent.put(std::move(outTracks_exgsf),instance);
+    iEvent.put(std::move(outTracks_rh),instance);
+
+}
+
+
+
+template <>
+void  TrackMergeremb<reco::MuonCollection >::willproduce(std::string instance, std::string alias)
+{
+   produces<reco::MuonCollection>();
+   produces<reco::CaloMuonCollection>();
+   produces<reco::MuonTimeExtraMap>("combined");
+   produces<reco::MuonTimeExtraMap>("dt");
+   produces<reco::MuonTimeExtraMap>("csc");
+  
+   // todo make this configurable (or not )
+   produces<reco::IsoDepositMap>("tracker");
+   produces<reco::IsoDepositMap>("ecal");
+   produces<reco::IsoDepositMap>("hcal");
+   produces<reco::IsoDepositMap>("ho");
+   produces<reco::IsoDepositMap>("jets");
+   
+   produces<reco::MuonToMuonMap>();
+   
+    
+}
+
+template <>
+void  TrackMergeremb<reco::MuonCollection >::willconsume(const edm::ParameterSet& iConfig)
+{
+
+   inputs_fixtrackrefs_ = consumes<TrackToTrackMapnew >( edm::InputTag("generalTracks") );
+   inputs_fixtrackcol_ = consumes<reco::TrackCollection >( edm::InputTag("generalTracks") );
+   
+   
+   
+}
+
+
+template <>
+void  TrackMergeremb<reco::MuonCollection  >::merg_and_put(edm::Event& iEvent, std::string instance,  std::vector<edm::EDGetTokenT<reco::MuonCollection> > &to_merge )
+{
+    std::unique_ptr<reco::MuonCollection> outTracks = std::unique_ptr<reco::MuonCollection>(new reco::MuonCollection);
+    std::unique_ptr<reco::CaloMuonCollection> calomu = std::unique_ptr<reco::CaloMuonCollection>(new reco::CaloMuonCollection); //not implemented so far 
+    
+    edm::Handle<TrackToTrackMapnew> track_ref_map;
+    iEvent.getByToken(inputs_fixtrackrefs_, track_ref_map);
+    
+    edm::Handle<reco::TrackCollection> track_new_col;
+    iEvent.getByToken(inputs_fixtrackcol_, track_new_col);   
+    std::map<reco::TrackRef , reco::TrackRef > simple_track_to_track_map; //I didn't find a more elegant way, so just build a good old fassion std::map
+     for (unsigned abc =0;  abc < track_new_col->size();  ++abc) {
+      reco::TrackRef trackRef(track_new_col, abc);      
+      simple_track_to_track_map[((*track_ref_map)[trackRef])[0]] = trackRef;
+    }
+    
+    std::vector<reco::MuonRef> muonRefColl;
+    reco::MuonRefProd outputMuonsRefProd = iEvent.getRefBeforePut<reco::MuonCollection>();
+    unsigned new_idx=0;
+    for (auto akt_collection : to_merge){
+        edm::Handle<reco::MuonCollection> track_col_in;  
+        iEvent.getByToken(akt_collection, track_col_in);
```

Comment:
this exercise of plain combination appears to be rather meaningless for overlapping muons (dR ~< 0.5)